--- ecmadmin.cpp~	2018-11-12 20:14:34.000000000 -0500
+++ ecmadmin.cpp	2018-11-12 20:14:40.000000000 -0500
@@ -112,7 +112,7 @@
             break;
 
          case 4:
-            strcpy(response, PromptForValue("Enter file to be imported: ", true));
+            strcpy(response, PromptForValue("Enter file to be imported: ", false));
 
             if (strlen(response) && OpenAdminSocket())
             {
