Index: WorkSender.cpp
===================================================================
--- WorkSender.cpp	(revision 2)
+++ WorkSender.cpp	(working copy)
@@ -342,7 +342,7 @@
    const char *updateSQL = "update Composite set LastSendTime = ? where CompositeName = ?;";
    const char *insertSQL = "insert into PendingWork ( UserID, MachineID, InstanceID, EmailID, CompositeName, " \
                            " FactorMethod, B1, Curves, DateAssigned ) " \
-                           " values ( ?,?,?,?,?,?,?,? ) ";
+                           " values ( ?,?,?,?,?,?,?,?,? ) ";
    bool success;
 
    if (!memcmp(ip_Socket->GetClientVersion(), "2.x", 3))
